### ISSUE TITLE
Fixture parameters marks work only if there's a single one marked

### ISSUE THREAD

**Issue by volnushka** [pytest-dev/pytest#12909]
Hi!
I'm trying to mark my fixtures' parameters to be able to invoke pytest with `-m` and make the tests apply a reduced parametrization. Here's a simple example of what I want to do:
```python
import pytest

@pytest.fixture(params=["opt1_notsmoke", pytest.param("opt1_smoke", marks=pytest.mark.smoke)])
def opt1(request):
    return request.param

@pytest.fixture(params=["opt2_notsmoke", pytest.param("opt2_smoke", marks=pytest.mark.smoke)])
def opt2(request):
    return request.param

def test_1(opt1, opt2):
    print(opt1)
    print(opt2)
```
However, this test is collected with the nodes that supposedly don't belong to the mark:
```bash
pytest pt.py --collect-only -m smoke

<Module pt.py>
    <Function test_1[opt1_notsmoke-opt2_smoke]>
    <Function test_1[opt1_smoke-opt2_notsmoke]>
    <Function test_1[opt1_smoke-opt2_smoke]>
```
But it works fine if I only mark parameter in a single fixture:
```python
@pytest.fixture(params=["opt1_notsmoke", pytest.param("opt1_smoke", marks=pytest.mark.smoke)])
def opt1(request):
    return request.param

@pytest.fixture(params=["opt2_without_mark"])
def opt2(request):
    return request.param
```
Collection:
```bash
pytest pt.py --collect-only -m smoke

<Module pt.py>
    <Function test_1[opt1_smoke-opt2_without_mark]>
```

I'd also posted a [question on StackOverflow](https://stackoverflow.com/questions/79100755/why-doesnt-marking-parameters-in-multiple-fixtures-work), but unfortunately no one answered that yet.
Thanks in advance!

Pytest `v.8.3.3`, also tried it on every version until `v.6.2.5`.
Pip packages:
```
Package              Version
-------------------- ---------
aenum                3.1.15
aiohappyeyeballs     2.4.0
aiohttp              3.10.6
aiosignal            1.3.1
async-timeout        4.0.3
attrs                24.2.0
certifi              2024.8.30
charset-normalizer   3.3.2
dill                 0.3.8
distro               1.9.0
exceptiongroup       1.2.2
frozenlist           1.4.1
idna                 3.10
iniconfig            2.0.0
multidict            6.1.0
packaging            24.1
pexpect              4.9.0
pip                  22.0.4
pluggy               1.5.0
ptyprocess           0.7.0
pytest               8.3.3
pytest-reportportal  5.4.2
pytest-rerunfailures 13.0
reportportal-client  5.5.7
requests             2.32.3
setuptools           56.0.0
tomli                2.0.1
typing_extensions    4.12.2
urllib3              2.2.3
yarl                 1.12.1
```

**Comment by The-Compiler** [comment]
Your example code seems to be missing your `opt2` fixture, FWIW.

**Comment by volnushka** [comment]
Yeah, thanks, added it.

**Comment by RonnyPfannschmidt** [comment]
Works as intended, parameters are a product,  so we get ⁴ tests

1 non marked
2 once marked
1 twice

**Comment by volnushka** [comment]
Ok, so am I using the wrong mechanism to apply a reduced parametrization of fixtures? Sorry for your time, but what is the dedicated one?

**Comment by RonnyPfannschmidt** [comment]
To answer ill use accumulate numbers 

⁴ tests total
3 tests of those with at least one mark
1 test of those 3 having 2 marks

If you only want the mark on the combination where both dimensions have the mark, it's currently necessary to manually write out the product 


If you provide more context I may be able to suggest something

**Comment by volnushka** [comment]
Sure. The goal is to add an invocation that it would run tests with a little bit altered parameter set. Lets's say, we have such fixtures:
```python
@pytest.fixture(params=["PEM", "DER"])
def cms_form(request):
    return request.param

@pytest.fixture(params=[False, True])
def okey_form_as_uri(request):
    return request.param

@pytest.fixture(params=[False, True])
def ocert_form_as_uri(request):
    return request.param

@pytest.fixture(params=[False, True])
def keyform_as_uri(request):
    return request.param

@pytest.fixture(params=[False, True])
def certform_as_uri(request):
    return request.param
```
These are used in many tests, so we chose to parametrize them via `fixture(params=...)` rather than via decorator, simply to have it in one place.
The idea is to add such an invocation parameter, that will collect only the desired parameter of each fixture (in this case - the first one), so that the result would be:
```bash
<Function test[PEM-False-False-False-False]>
```
In other words, we want to ommit some combinations for the sake of speed.

**Comment by RonnyPfannschmidt** [comment]
I'm not fluent in your problem domain, so the example without more context is not helpful 

At first glance I only have some ideas about dynamically declared plugin objects that would provide differently configured fixtures 

I'd like to avoi suggesting such a excessively tricky solution

**Comment by volnushka** [comment]
Ok, can't reveal the domain more than that, unfortunately. But you have certainly helped us, thanks!
